### PR TITLE
feat: move spline to helper in Visualization

### PR DESCRIPTION
**Pull request by asalzburger**
This PR moves the spline-interpolation from `Examples/Plugins/Obj` to `Core/Visualization` because it can be used in other visualisation areas as well.

I've added also a UnitTest since this is from some `experimetnal/usupported` Eigen directory, but it works nicely!

--- END COMMIT MESSAGE ---

Any further description goes here, @-mentions are ok here!

- Use a *conventional commits* prefix: [quick summary](https://www.conventionalcommits.org/en/v1.0.0/#summary)
  - We mostly use `feat`, `fix`, `refactor`, `docs`, `chore` and `build` types.
- A milestone will be assigned by one of the maintainers

## Summary by CodeRabbit

- **New Features**
	- Introduced a new interpolation method for 3D points using splines, enhancing the accuracy of data representation.
	- Expanded configurability of the `ObjSimHitWriter` with new parameters for trajectory smoothing and original hit retention.
	- Added unit tests for the new interpolation functionality to ensure reliability.

- **Bug Fixes**
	- Streamlined the interpolation process by replacing the previous method with a more efficient library function, improving maintainability.

- **Documentation**
	- Enhanced documentation for configuration parameters in the `ObjSimHitWriter` class, clarifying their purpose and usage.

snapshot01

### CODE REVIEW

File: Core/include/Acts/Visualization/Interpolation3D.hpp [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+/// @brief Helper function to interpolate points using a spline
+/// from Eigen
+///
+/// @tparam input_vector_type
+/// @param inputs input vector points
+/// @param nPoints number of interpolation points
+///
+/// @return std::vector<Acts::Vector3> interpolated points
+template <typename input_vector_type>
+std::vector<Acts::Vector3> spline(const std::vector<input_vector_type>& inputs,
```
Comment (andiwand) [inline Core/include/Acts/Visualization/Interpolation3D.hpp:26]: I wonder if we should fully template this on a range to decouple from `std::vector`? Also do we plan to use this with something else than `Vector3` as input?

File: Core/include/Acts/Visualization/Interpolation3D.hpp [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+  if (inputs.empty()) {
+    return output;
+  }
+
+  if (nPoints < 2) {
+    // No interpolation done return simply the output vector
+    for (const auto& input : inputs) {
+      output.push_back(input.template head<3>());
+    }
```
Comment (andiwand) [inline Core/include/Acts/Visualization/Interpolation3D.hpp:38]: potentially an error case

File: Core/include/Acts/Visualization/Interpolation3D.hpp [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+///
+/// @tparam input_vector_type
+/// @param inputs input vector points
+/// @param nPoints number of interpolation points
+///
+/// @return std::vector<Acts::Vector3> interpolated points
+template <typename input_vector_type>
+std::vector<Acts::Vector3> spline(const std::vector<input_vector_type>& inputs,
+                                  std::size_t nPoints) {
+  std::vector<Acts::Vector3> output;
```
Comment (andiwand) [inline Core/include/Acts/Visualization/Interpolation3D.hpp:28]: we could reserve on `output` after we know that the interpolation will go through

File: Core/include/Acts/Visualization/Interpolation3D.hpp [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/Algebra.hpp"
+
+#include <unsupported/Eigen/Splines>
```
Comment (andiwand) [inline Core/include/Acts/Visualization/Interpolation3D.hpp:13]: that looks a bit scary

File: Core/include/Acts/Visualization/Interpolation3D.hpp (hunk 1 above)
Comment (asalzburger) [reply Core/include/Acts/Visualization/Interpolation3D.hpp]: Absolutely, if I want to user it with the actsvg plugin, I will need another input/output type.

I made it agnostic to it.

File: Core/include/Acts/Visualization/Interpolation3D.hpp (hunk 3 above)
Comment (asalzburger) [reply Core/include/Acts/Visualization/Interpolation3D.hpp]: yes, I've changed that in the meantime.

**Review by andiwand — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results [a646e0655aa429c3a8ef79d11a0ed5b2f335ce0a]
### :white_check_mark: All tests successful

|status|job|report|
|------|---|------|
|:green_circle: | [report_pull_request](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47647466)||
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47647465)||
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47647464)||
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47647463)||
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47647462)||
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47647461)||
|:green_circle: | [test_ActsDumpGeometryIdentifiers](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47647460)||
|:green_circle: | [test_ActsCheckObjectCountsWorkflowHgtd](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47647459)||
|:green_circle: | [test_ActsCheckObjectCountsCachedWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47647458)||
|:green_circle: | [test_ActsCheckObjectCountsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47647457)||
|:green_circle: | [test_ActsEFTrackFit](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47647456)||
|:green_circle: | [test_ActsPersistifySeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47647455)||
|:green_circle: | [test_ActsBenchmarkWithSpot](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47647454)||
|:green_circle: | [test_ActsAnalogueClustering](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47647453)||
|:green_circle: | [test_ActsWorkflowHeavyIons](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47647452)||
|:green_circle: | [test_ActsWorkflowFastTracking](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47647451)||
|:green_circle: | [test_ActsWorkflowCached](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47647450)||
|:green_circle: | [test_ActsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47647449)||
|:green_circle: | [test_ActsValidateAmbiguityResolution](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47647448)||
|:green_circle: | [test_ActsValidateResolvedTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47647447)||
|:green_circle: | [test_ActsValidateTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47647446)||
|:green_circle: | [test_ActsValidateActsCoreSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47647445)||
|:green_circle: | [test_ActsValidateActsSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47647444)||
|:green_circle: | [test_ActsValidateSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47647443)||
|:green_circle: | [test_ActsValidateOrthogonalSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47647442)||
|:green_circle: | [test_ActsValidateClusters](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47647441)||
|:green_circle: | [test_ActsPersistifyEDM](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47647440)||
|:green_circle: | [test_ActsGx2fRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47647439)||
|:green_circle: | [test_ActsGSFRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47647438)||
|:green_circle: | [test_ActsKfRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47647437)||
|:green_circle: | [test_ActsExtrapolationAlgTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47647436)||
|:green_circle: | [test_ActsITkTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/47647435)||

**Pull request merged**